cloudstack/loadbalancer.go

```diff
@@ -346,19 +346,29 @@ func getLoadBalancerRule(client *cloudstack.CloudStackClient, lbName, projectID
 		Count             int                 `json:"count"`
 		LoadBalancerRules []*loadBalancerRule `json:"loadbalancerrule"`
 	}
+	var lbResult *loadBalancerRule
 
 	err := client.Custom.CustomRequest("listLoadBalancerRules", pc, &result)
 	if err != nil {
 		return nil, err
 	}
-	if result.Count > 1 {
+	if result.Count == 0 {
+		return nil, nil
+	}
+	count := 0
+	for idx := range result.LoadBalancerRules {
```

Comment:
```suggestion
	for _, lbRule := range result.LoadBalancerRules {
```

So you will can compare their names with `if lbRule.Name == lbName`.